### ISSUE TITLE
When TP and CP are mixed the loss is reduced incorrectly

### ISSUE THREAD

**Issue by francesco-bertolotti** [pytorch/torchtitan#3329]
There appears to be an issue with how the *displayed* loss (not the training loss) is computed when Context Parallelism (CP) and Tensor Parallelism (TP) are both enabled.

I believe the problem originates in this method:

https://github.com/pytorch/torchtitan/blob/7f602b98887b52fa3d643f2f50c18ea7af33c875/torchtitan/distributed/utils.py#L30-L72

When TP is enabled, the loss becomes a `DTensor`. This triggers the early return at line 64, meaning the value is never reduced across CP ranks. As a result, configurations such as `TP=2` and `CP=2` produce a displayed loss that is roughly 2× smaller than the actual value.

I ran a few experiments locally by removing the early `return` and instead converting the tensor with:

```python
x = float(x.full_tensor().item())
```
This seems to produce the expected behavior in my tests.

Since this looks like a sensitive point in the codebase, I wanted to open an issue first before preparing a PR.

**Comment by francesco-bertolotti** [comment]
Closing since a similar issue was opened in
https://github.com/pytorch/torchtitan/issues/3280
and PR is in progress in 
https://github.com/pytorch/torchtitan/pull/3159
